src/ZPublisher/BaseRequest.py

```diff
@@ -502,6 +490,7 @@ def traverse(self, path, response=None, validated_hook=None):
                             self.roles = getRoles(
                                 object, '__call__',
                                 object.__call__, self.roles)
+                            self.ensure_publishable(object.__call__, True)
```

Comment:
This addition represents a behavior change, there was no such check in that part of the code before. Were you just correcting a long-standing omission or did you encounter specific problems?